bin2tif/terra_bin2tif.py

```diff
@@ -80,81 +83,169 @@ def process_message(self, connector, host, secret_key, resource, parameters):
                                               timestamp[:4], timestamp[5:7], timestamp[8:10],
-                                              leaf_ds_name=self.sensors.get_display_name()+' - '+timestamp)
+                                              leaf_ds_name=self.sensors.get_display_name() + ' - ' + timestamp)
 
-        # Upload original Lemnatec metadata to new Level_1 dataset
-        md = get_terraref_metadata(all_dsmd)
-        md['raw_data_source'] = host + ("" if host.endswith("/") else "/") + "datasets/" + resource['id']
-        lemna_md = build_metadata(host, self.extractor_info, target_dsid, md, 'dataset')
-        self.log_info(resource, "uploading LemnaTec metadata")
-        upload_metadata(connector, host, secret_key, target_dsid, lemna_md)
+        uploaded_file_ids = []
 
+        self.log_info(resource, "uploading LemnaTec metadata to ds [%s]" % target_dsid)
+        remove_metadata(connector, host, secret_key, target_dsid, self.extractor_info['name'])
+        terra_md_trim = get_terraref_metadata(all_dsmd)
+        if updated_experiment:
+            terra_md_trim['experiment_metadata'] = expmd
+        terra_md_trim['raw_data_source'] = host + ("" if host.endswith("/") else "/") + "datasets/" + resource['id']
+        level1_md = build_metadata(host, self.extractor_info, target_dsid, terra_md_trim, 'dataset')
+        upload_metadata(connector, host, secret_key, target_dsid, level1_md)
+
+        left_shape = terraref.stereo_rgb.get_image_shape(metadata, 'left')
+        right_shape = terraref.stereo_rgb.get_image_shape(metadata, 'right')
+        gps_bounds = geojson_to_tuples(terra_md_full['spatial_metadata']['stereoTop']['bounding_box'])
 
         if (not file_exists(left_tiff)) or self.overwrite:
             self.log_info(resource, "creating & uploading %s" % left_tiff)
             left_image = terraref.stereo_rgb.process_raw(left_shape, img_left, None)
+            out_tmp_tiff_left = os.path.join(tempfile.gettempdir(), resource['id'].encode('utf8'))
+             # Rename output.tif after creation to avoid long path errors
+            create_geotiff(left_image, gps_bounds, out_tmp_tiff_left, None, True, self.extractor_info, terra_md_full)
 
-            # Rename output.tif after creation to avoid long path errors
-            create_geotiff(left_image, left_gps_bounds, out_tmp_tiff, None, False, self.extractor_info, metadata)
             # TODO: we're moving zero byte files
-            shutil.move(out_tmp_tiff, left_tiff)
-            if left_tiff not in resource['local_paths']:
-                fileid = upload_to_dataset(connector, host, self.clowder_user, self.clowder_pass, target_dsid, left_tiff)
-                uploaded_file_ids.append(host + ("" if host.endswith("/") else "/") + "files/" + fileid)
+            if os.path.getsize(out_tmp_tiff_left) > 0:
+                shutil.move(out_tmp_tiff_left, left_tiff)
+                if left_tiff not in resource['local_paths']:
```

Comment:
here let's use the found_in_dataset() model:
https://github.com/terraref/extractors-multispectral/pull/32/files#diff-a186da4a4378b6d0a96e0c8abd3bc92bR158

Reply:
fixed